src/main/java/harvard/Parser.java

```diff
@@ -0,0 +1,127 @@
+package harvard;
+
+import java.time.LocalDate;
+
+import harvard.exceptions.HarvardException;
+import harvard.tasks.Deadline;
+import harvard.tasks.Event;
+import harvard.tasks.Task;
+import harvard.tasks.Todo;
+
+/**
+ * The Parser class is responsible for interpreting user commands in the Harvard application.
+ * It processes user input and performs corresponding actions such as adding tasks, deleting tasks,
+ * marking tasks as done, and displaying the task list.
+ */
+public class Parser {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs a Parser instance with references to Storage, TaskList, and Ui.
+     *
+     * @param s The Storage instance for storing tasks.
+     * @param tL The TaskList instance for managing tasks.
+     * @param ui The Ui instance for displaying messages.
+     */
+    public Parser(Storage s, TaskList tL, Ui ui) {
+        this.storage = s;
+        this.tasks = tL;
+        this.ui = ui;
+    }
+
+    /**
+     * Parses the user command and performs corresponding actions.
+     *
+     * @param commandLine The user command to be parsed.
+     * @throws HarvardException If an error occurs during parsing or task manipulation.
+     */
+    public String parse(String commandLine) throws HarvardException {
+        String command = commandLine.split(" ")[0];
+
+        if (!command.equals("list") && !command.equals("todo")
```

Comment:
Would use of switch-case be better here? Any reason you went for if-else rather than switch-case?